### PR TITLE
[terra-clinical-onset-picker] Added upgrade guide for v3 to v4

### LINKED ISSUE #604: [onset-picker] Missing v3 to v4 Upgrade Guide
# Bug Report

## Description
Missing an upgrade guide for v3 to v4 for terra-clinical-onset-picker.

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Fixes #604

### CODE REVIEW

File: packages/terra-clinical-onset-picker/docs/UPGRADEGUIDE.md [hunk 1]
```diff
@@ -1,5 +1,25 @@
 # Terra Clinical Onset Picker Upgrade Guide
 
+## Changes from 3.0 to 4.0
+
+### Breaking Change
+* Condensed onChange function props into a single prop (`onsetOnChange`) that returns an Object for the onsetDate.
+
+## New Feature
+
+* Added support for age units.
```
Comment (emilyrohrbough) [inline packages/terra-clinical-onset-picker/docs/UPGRADEGUIDE.md]: What are the added support? How does someone leverage this?

File: packages/terra-clinical-onset-picker/docs/UPGRADEGUIDE.md [hunk 2]
```diff
@@ -1,5 +1,25 @@
 # Terra Clinical Onset Picker Upgrade Guide
 
+## Changes from 3.0 to 4.0
+
+### Breaking Change
+* Condensed onChange function props into a single prop (`onsetOnChange`) that returns an Object for the onsetDate.
```
Comment (emilyrohrbough) [inline packages/terra-clinical-onset-picker/docs/UPGRADEGUIDE.md]: What changes would one expect with this prop change?

File: packages/terra-clinical-onset-picker/docs/UPGRADEGUIDE.md [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
+
+## New Feature
+
+* Added support for age units.
+
+### Removed Props
+
+* Removed prop `granularitySelectOnChange`
+* Removed prop `precisionSelectOnChange`
+* Removed prop `onsetDateInputOnChange`
```
Comment (emilyrohrbough) [inline packages/terra-clinical-onset-picker/docs/UPGRADEGUIDE.md]: Shouldn't this be listed under the Breaking change with instructures on how to convert to use onsetOnChange?

Comment (saket2403) [reply packages/terra-clinical-onset-picker/docs/UPGRADEGUIDE.md]: Moved to breaking 77f1bc1

File: packages/terra-clinical-onset-picker/docs/UPGRADEGUIDE.md (hunk 2 above)
Comment (saket2403) [reply packages/terra-clinical-onset-picker/docs/UPGRADEGUIDE.md]: Added code diff sample 77f1bc1

File: packages/terra-clinical-onset-picker/docs/UPGRADEGUIDE.md (hunk 1 above)
Comment (saket2403) [reply packages/terra-clinical-onset-picker/docs/UPGRADEGUIDE.md]: Added more details 77f1bc1

**Pull request merged**